rest/config.go

```diff
@@ -676,6 +677,38 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 	return nil
 }
 
+// expandEnv replaces $var or ${var} in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by an empty string. A default value can
+// be given by using the form ${var:-default value}.
+func expandEnv(config []byte) []byte {
+	return []byte(os.Expand(string(config), func(key string) string {
+		if key == "$" {
+			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
+			return key
+		}
+		return envDefaultExpansion(key, os.Getenv)
```

Comment:
I forgot to mention before, but the benefit of passing an `os.Getenv` function into this, aside from just testing, is that it can be adapted to read our proposed ".env" file :) not just directly call `os.Getenv`.